packages/core/src/components/forms/numericInput.tsx

```diff
@@ -533,37 +419,35 @@ export class NumericInput extends AbstractPureComponent<HTMLInputProps & INumeri
         Utils.safeInvoke(this.props.onKeyDown, e);
     };
 
-    private handleInputKeyPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    private handleInputKeyPress = (e: React.KeyboardEvent) => {
         // we prohibit keystrokes in onKeyPress instead of onKeyDown, because
         // e.key is not trustworthy in onKeyDown in all browsers.
-        if (this.props.allowNumericCharactersOnly && this.isKeyboardEventDisabledForBasicNumericEntry(e)) {
+        if (this.props.allowNumericCharactersOnly && !isValidNumericKeyboardEvent(e)) {
             e.preventDefault();
         }
 
         Utils.safeInvoke(this.props.onKeyPress, e);
     };
 
-    private handleInputPaste = (e: React.ClipboardEvent<HTMLInputElement>) => {
+    private handleInputPaste = (e: React.ClipboardEvent) => {
         this.didPasteEventJustOccur = true;
         Utils.safeInvoke(this.props.onPaste, e);
     };
 
-    private handleInputChange = (e: React.FormEvent<HTMLInputElement>) => {
+    private handleInputChange = (e: React.FormEvent) => {
         const value = (e.target as HTMLInputElement).value;
 
         let nextValue: string;
 
         if (this.props.allowNumericCharactersOnly && this.didPasteEventJustOccur) {
             this.didPasteEventJustOccur = false;
             const valueChars = value.split("");
-            const sanitizedValueChars = valueChars.filter(this.isFloatingPointNumericCharacter);
+            const sanitizedValueChars = valueChars.filter(isFloatingPointNumericCharacter);
             const sanitizedValue = sanitizedValueChars.join("");
             nextValue = sanitizedValue;
         } else {
             nextValue = value;
         }
-
-        this.shouldSelectAfterUpdate = false;
```

Comment:
Just checking: do we not need to explicitly set this to `false` via `this.setState` here?

Reply:
mm yes probably will do

Reply:
nice catch!!